doc/source/ray-air/examples/pytorch_resnet_finetune.ipynb

```diff
@@ -0,0 +1,1152 @@
+    "\n",
+    "            # Create a dataset iterator for the shard on the current worker\n",
+    "            dataset_iterator = datasets[phase].iter_torch_batches(\n",
+    "                batch_size=worker_batch_size, device=device\n",
+    "            )\n",
+    "            for batch in dataset_iterator:\n",
+    "                inputs = batch[\"image\"]\n",
+    "                labels = batch[\"label\"]\n",
+    "\n",
+    "                # zero the parameter gradients\n",
+    "                optimizer.zero_grad()\n",
+    "\n",
+    "                # forward\n",
+    "                with torch.set_grad_enabled(phase == \"train\"):\n",
+    "                    # Get model outputs and calculate loss\n",
+    "                    outputs = model(inputs)\n",
+    "                    loss = criterion(outputs, labels)\n",
+    "\n",
+    "                    # backward + optimize only if in training phase\n",
+    "                    if phase == \"train\":\n",
+    "                        loss.backward()\n",
+    "                        optimizer.step()\n",
+    "\n",
+    "                # calculate statistics\n",
+    "                running_loss += loss.item() * inputs.size(0)\n",
+    "                running_corrects += evaluate(outputs, labels)\n",
+    "\n",
+    "            epoch_loss = running_loss / datasets[phase].count()\n",
+    "            epoch_acc = running_corrects / datasets[phase].count()\n",
+    "\n",
+    "            if session.get_world_rank() == 0:\n",
+    "                print(\n",
+    "                    \"Epoch {}-{} Loss: {:.4f} Acc: {:.4f}\".format(\n",
+    "                        epoch, phase, epoch_loss, epoch_acc\n",
+    "                    )\n",
+    "                )\n",
+    "\n",
+    "            # Report metrics and checkpoint every epoch\n",
+    "            if phase == \"val\":\n",
+    "                checkpoint = TorchCheckpoint.from_dict(\n",
+    "                    {\n",
+    "                        \"epoch\": epoch,\n",
+    "                        \"model\": model.module.state_dict(),\n",
+    "                        \"optimizer_state_dict\": optimizer.state_dict(),\n",
+    "                    }\n",
+    "                )\n",
+    "                session.report(\n",
+    "                    metrics={\"loss\": epoch_loss, \"acc\": epoch_acc},\n",
+    "                    checkpoint=checkpoint,\n",
+    "                )\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "VmZQKtWnrLT3"
+   },
+   "source": [
+    "Next, setup the TorchTrainer:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {
+    "cellView": "form",
+    "id": "6jeTo6YTrLT3",
+    "tags": [
+     "remove-cell"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "# @title +\n",
+    "if SMOKE_TEST:\n",
+    "    from moto.server import ThreadedMotoServer\n",
+    "    from moto import mock_s3\n",
```

Comment:
yeah i don't think we should have this part here as part of the example

Reply:
This indeed looks messy, but we make the codes in hidden cells so the users won't see this in doc.

I thought it would be easy to just pass a S3 uri to syncConfig to save checkpoints. But actually our ci workers are not allowed to upload files to real S3. So I wrote mock S3 server here.

Do you think it's better to just remove the S3 things, and use a local_dir instead?